### PR TITLE
Replace guava's Optional with java.util's one

### PR DESCRIPTION
**Pull request by patrox**

### CODE REVIEW

**Comment by jplock** [comment]
This is great, thank you!  Any feedback from @dropwizard/committers about this?  This may break some things, but probably worthwhile to get into 1.0.0-rc2.

File: dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public final void run(T configuration, Environment environment) throws Exception
         environment.healthChecks().register(name(),
                                             new SessionFactoryHealthCheck(
                                                     environment.getHealthCheckExecutorService(),
-                                                    dbConfig.getValidationQueryTimeout().or(Duration.seconds(5)),
+                                                    dbConfig.getValidationQueryTimeout().orElseGet(() -> Duration.seconds(5)),
```
Comment (nickbabcock) [inline dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java]: Why not `orElse(Duration.seconds(5))`?

Comment (patrox) [inline dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java]: To avoid evaluating it, if it's not needed.

Comment (joschi) [inline dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java]: This is being called *once*, so I don't think that performance optimization is relevant here.

**Comment by joschi** [comment]
LGTM except for that minor comment on `HibernateBundle`. :+1:

File: dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java (hunk 1 above)
Comment (patrox) [inline dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java]: force of habit - will change that

**Comment by joschi** [comment]
Thanks!

**Pull request merged**